src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentLoaderService.cs

```diff
@@ -167,11 +167,15 @@ internal sealed class PdbSourceDocumentLoaderService : IPdbSourceDocumentLoaderS
             return null;
         }
 
-        private SourceFileInfo? TryGetOriginalFile(SourceDocument sourceDocument, Encoding encoding, TelemetryMessage telemetry)
+        private SourceFileInfo? TryGetOriginalFile(string tempFilePath, SourceDocument sourceDocument, Encoding encoding, TelemetryMessage telemetry)
         {
             if (File.Exists(sourceDocument.FilePath))
             {
-                var result = LoadSourceFile(sourceDocument.FilePath, sourceDocument, encoding, FeaturesResources.external, ignoreChecksum: false);
+                // Copy the source file to a temp path so it can't change, making it invalid, while open in VS
+                var tempSourcePath = Path.Combine(tempFilePath, Path.GetFileName(sourceDocument.FilePath));
```

Comment:
If I have two different files with the same file name (but different paths) in my PDB, is this going to stomp over the same file?

Reply:
Would be good idea to include checksum in the path (subdirectory).

Reply:
Or recreate the directory structure of the project.